WcaOnRails/app/controllers/api/v0/user_roles_controller.rb

```diff
@@ -149,7 +149,14 @@ def index_for_group_type
     # Temporary hack to support the old system roles, will be removed once all roles are
     # migrated to the new system.
     if group_type == UserGroup.group_types[:delegate_regions]
-      roles.concat(User.where.not(delegate_status: nil).map(&:delegate_role))
+      # extra_metadata adds some extra metadata to the roles. These metadata computation is bit costly,
+      # so we only compute it when extra_metadata is true.
+      extra_metadata = params.key?(:extraMetadata) ? ActiveRecord::Type::Boolean.new.cast(params.require(:extraMetadata)) : nil
+      if extra_metadata
+        roles.concat(User.where.not(delegate_status: nil).map(&:delegate_role_with_extra_metadata))
+      else
+        roles.concat(User.where.not(delegate_status: nil).map(&:delegate_role))
+      end
```

Comment:
Why are you doing `User.where.not(delegate_status: nil)` here, instead of using the available scope `User.delegates`?